### PR TITLE
Back end user

### PR DESCRIPTION
**Pull request by boyajay**
#### What's this PR do?

1) Add functions to controller: getUser and getFriends
2) Modulated out functions with DB interactions to the model file.
#### Any background context you want to provide?

factored out DB interactions from server interactions for ease of testing
#### What are the relevant tickets/issues?

none
#### Where should the reviewer start?

users.js, usersController.js
#### Are there tests written yet? How should this be manually tested?

Ready for tests to be written
#### Questions:
- Does this add new dependencies?
  no
- Does our documentation need to be updated?
  no

### CODE REVIEW

File: server/users/users.js [hunk 1]
```diff
@@ -15,6 +16,59 @@ const UserFriend = db.define('friends', {});
 // will add methods to User (ex. User.getFriends())
 User.belongsToMany(User, { as: 'Friends', through: UserFriend });
 
+User.getUserInfo = fbId => {
+  User.find({ where: { fbId } })
```
Comment (abuddiga) [inline server/users/users.js]: can use findOne for faster lookup

File: server/users/users.js [hunk 2]
```diff
@@ -15,6 +16,59 @@ const UserFriend = db.define('friends', {});
...
+      const response = {
+        user,
+        token,
+      };
+      return response;
+    })
+    .catch(err => err);
+};
+
+User.getUserFriends = userId => {
```
Comment (abuddiga) [inline server/users/users.js]: don't need this function--we can just use user.getFriends

File: server/users/users.js [hunk 3]
```diff
@@ -15,6 +16,52 @@ const UserFriend = db.define('friends', {});
...
+        };
+        return User.create(newUser);
+      }
+      // if user already exists, update user entry in the database
+      const updatedInfo = {
+        name,
+        picture,
+      };
+      return match.update(updatedInfo)
+      .then(data => data)
```
Comment (abuddiga) [inline server/users/users.js]: Don't need these then and catches--it'll return the result of the update automatically and line 60 will catch

File: server/config/routes.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ const Strategy = require('passport-facebook').Strategy;
 
 module.exports = function (app, express) {
   app.get('api/me'/* , controller function here */);
-  app.get('/api/users/:user_id/friends'/* , controller function here */);
+  app.get('/api/users/me/friends'/* , controller function here */);
```
Comment (abuddiga) [inline server/config/routes.js]: api/me/friends

**Comment by xuejinglu** [comment]
:100: :+1: :+1:

**Comment by abuddiga** [comment]
:100: :8ball:

**Comment by leranf** [comment]
yaaa buddyyyy

**Pull request merged**